lib/src/components/NxSearchDropdown/stateful/__tests__/NxStatefulSearchDropdown.test.tsx

```diff
@@ -4,85 +4,349 @@
+        .toHaveAttribute('aria-hidden', 'true');
+    expect(quickRender({ defaultSearchText: 'foo', disabled: false }).getByRole('alert'))
+        .toHaveAttribute('aria-hidden', 'false');
+    expect(quickRender({ defaultSearchText: 'foo', disabled: undefined }).getByRole('alert'))
+        .toHaveAttribute('aria-hidden', 'false');
+    expect(quickRender({ defaultSearchText: 'foo', disabled: null }).getByRole('alert'))
+        .toHaveAttribute('aria-hidden', 'false');
+    expect(quickRender({ defaultSearchText: 'foo' }).getByRole('alert'))
+        .toHaveAttribute('aria-hidden', 'false');
+  });
+
+  it('sets aria-live on the dropdown menu to "polite"', function() {
+    expect(quickRender().getByRole('alert', { hidden: true })).toHaveAttribute('aria-live', 'polite');
+  });
+
+  it('sets aria-busy on the dropdown menu if loading is true', function() {
+    expect(quickRender().getByRole('alert', { hidden: true })).toHaveAttribute('aria-busy', 'false');
+    expect(quickRender({ loading: true }).getByRole('alert', { hidden: true })).toHaveAttribute('aria-busy', 'true');
+  });
+
+  it('sets the alert role on the dropdown menu when it is in loading, error, or empty states', function() {
+    expect(quickRender({ defaultSearchText: 'asdf', matches: [] }).getByRole('alert')).toBeInTheDocument();
+    expect(quickRender({ defaultSearchText: 'asdf', loading: true }).getByRole('alert')).toBeInTheDocument();
+
+    const dropdownMenu = quickRender({ defaultSearchText: 'asdf', error: 'foo' }).getAllByRole('alert')[0];
+    expect(dropdownMenu).toBeInTheDocument();
+
+    const errorAlert = within(dropdownMenu).getByRole('alert');
+    expect(errorAlert).toBeInTheDocument();
+  });
+
+  it('sets an id on the dropdown and references it in the search box\'s aria-controls', function() {
+    const el = quickRender({ defaultSearchText: 'foo' }),
+        filterInput = el.getByRole('searchbox'),
+        dropdown = el.getByRole('alert');
+
+    expect(dropdown).toHaveAttribute('id');
+    expect(filterInput).toHaveAttribute('aria-controls', dropdown.getAttribute('id'));
+  });
+
+  it('sets aria-haspopup on the filter input', function() {
+    const el = quickRender({ defaultSearchText: 'foo' }),
+        filterInput = el.getByRole('searchbox');
+
+    expect(filterInput).toHaveAttribute('aria-haspopup', 'menu');
+  });
+
+  it('shows loading text on the dropdown when loading prop is true', function() {
+    const el = quickRender({ defaultSearchText: 'foo', loading: true });
+
+    expect(el.getByRole('status')).toHaveTextContent('Loading…');
+  });
+
+  it('shows error text when the error prop is provided', function() {
+    const el = quickRender({ defaultSearchText: 'foo', error: 'bar' });
+
+    const dropdownMenu = el.getAllByRole('alert')[0];
+    expect(within(dropdownMenu).getByRole('alert')).toHaveTextContent('bar');
+  });
+
+  it('fires onSearch with the defaultSearchText when the load wrappers retryHandler is triggered', async function() {
```

Comment:
It should fire it with the current text input value, which isn't necessarily `defaultSearchText`

Reply:
> It should fire it with the current (trimmed) text input value, which isn't necessarily `defaultSearchText`

Updated here: https://github.com/sonatype/sonatype-react-shared-components/pull/933/commits/aa5f41008258e8af27b6dc38fb812a28a69195f4#diff-90808a1737028207ca6500c1a317cec13f908eeab0b3fa61ae10a1bc93e9f19cR223-R244